lib/puppet/provider/bootstrap_rpm/bootstrap_rpm.rb

```diff
@@ -0,0 +1,139 @@
+require 'fileutils'
+
+Puppet::Type.type(:bootstrap_rpm).provide(:bootstrap_rpm) do
+
+  commands :rpmbuild => 'rpmbuild'
+
+  def create
+    base_dir
+    copy_sources
+    write_specfile
+    build_rpm
+    copy_rpm
+    link_rpm
+  ensure
+    FileUtils.remove_dir(base_dir) if File.directory?(base_dir)
+  end
+
+  def delete
+  end
+
+  def exists?
+    File.exist?(resource[:symlink]) if resource[:symlink]
+  end
+
+	private
+
+  def base_dir
+    @base_dir ||= Dir.mktmpdir
+  end
+
+  def sources_dir
+    mkdir('SOURCES')
+  end
+
+  def build_dir
+    mkdir('BUILD')
+  end
+
+  def spec_dir
+    mkdir('SPECS')
+  end
+
+  def rpm_dir
+    mkdir('RPMS')
+  end
+
+  def copy_sources
+    FileUtils.copy(resource[:script], "#{sources_dir}/#{File.basename(resource[:script])}")
+  end
+
+  def write_specfile
+    File.open("#{spec_dir}/#{resource[:name]}.spec", 'w') do |file|
+      file.write(spec)
+    end
+  end
+
+  def release
+    if !latest_rpm.nil? && File.exist?(latest_rpm)
+      rel = `rpm -qp #{latest_rpm} --queryformat=%{release}`
```

Comment:
This is not how you should call commands. This should be done in the same way as `commands :rpmbuild`. Then you call it as `rpm('-qp', latest_rpm, '--queryformat=%{release}')`